### DISCUSSION TITLE
v0.3.3

### DISCUSSION THREAD

**Discussion by stefansundin** [stefansundin/privatkopiera#70] [Announcements]
**Hej! Jag efterlyser någon som är bra på videoproduktion och vill göra en instruktionsfilm till Privatkopiera. För mer information, se https://github.com/stefansundin/privatkopiera/discussions/63.**

Ändringar i v0.3.3 ([kodändringar](https://github.com/stefansundin/privatkopiera/compare/v0.3.2...v0.3.3)):

- Fixade svtplay.

Diskutera nedan.

**Obs:** om du har ett problem som du tror beror på koden och du vill testa en äldre version så kan du, om du använder Firefox, installera en gammal version av tillägget [på den här sidan](https://addons.mozilla.org/en-US/firefox/addon/privatkopiera/versions/). Det finns tyvärr inget enkelt sätt att göra detsamma i Chrome. **Detta hade inte löst problemet den här gången dock.**

**Comment by stefansundin** [comment]
Jag har pillat lite mer med svtplay efter att jag släppte v0.3.3, och lyckades klura ut hur deras graphql grej fungerar. Det är nog mycket mycket mer pålitligt än att parsa HTML.

Nästa gång svtplay slutar fungera så ska jag uppdatera till detta.

Exempel: https://www.svtplay.se/video/2520376/pippi-langstrump/pippi-langstrump-sasong-1-avsnitt-1
ID: 2520376

```
$ curl https://api.svt.se/contento/graphql --data-raw '{"query":"query VideoPage($legacyIds: [Int!]!) { listablesByEscenicId(escenicIds: $legacyIds) { videoSvtId }}", "variables": { "legacyIds": [2520376] }}'
{"data":{"listablesByEscenicId":[{"videoSvtId":"jBDqx98"}]}}
```
https://api.svt.se/video/jBDqx98

🎉

**Reply by starwars74** [reply]
Tyvärr har urplay.se har slutat fungera.

**Reply by stefansundin** [reply]
Fick en rapport i #75. Upplever du samma problem? Kan du länka video med problem?

**Reply by starwars74** [reply]
Ja, samma problem. Ett ex https://urplay.se/program/215700-pa-kna-i-protest

**Reply by stefansundin** [reply]
Ok. Släpper nog en ny version med fix imorgon.

**Reply by starwars74** [reply]
Ok. Tack

**Reply by stefansundin** [reply]
v0.3.4 har nu laddats upp med en fix.

Observera att Chrome har en obligatorisk granskningsprocess för nya versioner. Därför får Chrome uppdateringen först om några dagar.

Uppdateringen borde redan vara tillgänglig i Firefox. Om du fortfarande har den gamla versionen så kan du uppdatera dina tillägg med en knapp på addon-sidan där listan med dina tillägg syns ([se bild](https://user-images.githubusercontent.com/1991151/108611585-c5740080-7394-11eb-8cb6-0a850d19a1b7.png)).

**Reply by starwars74** [reply]
Ok. Tack!

**Comment by MeSwede** [comment]
Hej!

Jag har laddat ner ditt program... Samt FFmpeg-filerna.

Inte riktigt klart vad som saknas på min sida... Vore tacksam för ytterligar information och instruktioner!

Tack!

MeSwede

**Reply by stefansundin** [reply]
Har du följt instruktionerna som är [längst upp på hemsidan](https://stefansundin.github.io/privatkopiera/)?

> Följ dessa steg:
> 1. Hitta en video på svtplay.se som du vill ladda ned.
> 2. Öppna Privatkopiera genom att klicka på VHS-ikonen till höger om addressfältet.
> 3. Vänta några sekunder tills tillägget hittar strömmen.
> 4. Klicka på knappen Kopiera kommando.
> 5. Klistra in kommandot i en terminal (t.ex. `cmd.exe` i Windows, använd högerklick för att klistra in). Tryck enter.
> 6. Förhoppningsvis har du filen nedladdad efter några minuter.

Vad är det som är oklart?

**Reply by MeSwede** [reply]
Jag har bara en Mac-dator för tillfället...

Öppnar TERMINAL på dators och paste:ar in den kopierade kommandotexten...

...inget händer!

Verkar som det är två stökar i kommando-texten... Dessa genererar två filer - en audio-only och en video-only - i mitt gamla downloading-of-streaming-program...

Så filerna är identifierade - men saknar slutspurten!

Tack igen!

MeSwede

**Comment by MeSwede** [comment]
...får tag på audiofilerna i nedladdad fil, dock ej videofilerna!

Tack igen!

MeSwede

**Comment by MeSwede** [comment]
...får tag på audio- och videofilerna i nedladdningen - dock som separata filer!

Tack igen (för hjälp)!

MeSwede

**Comment by trekronor** [comment]
> Ändringar i v0.3.3 ([kodändringar](https://github.com/stefansundin/privatkopiera/compare/v0.3.2...v0.3.3)):
> 
>     * Fixade svtplay.

Tack för fixen! En sak, att undertexter nu inkluderas i kommandoraden och därmed i mkv-filen per default kan man på något sätt få slippa det i en kommande version? En switch kanske så alla blir glada :)

**Reply by Bub0** [reply]
Se https://github.com/stefansundin/privatkopiera/discussions/64#discussioncomment-338498

**Reply by stefansundin** [reply]
Du måste manuellt ändra filändelsen på till `.mp4` så blir det som förut. Tyvärr går det för tillfället inte att ställa in en default som gör detta automatiskt.

**Reply by trekronor** [reply]
Okej bra att veta. Gick ju även bra att bara klippa bort textnings-koden i kommandoraden och kopiera koden som vanligt.

**Comment by ingemar1961** [comment]
Jag har donerat 15USD och skulle önska en möjlighet att ladda ner från dr.dk/tv.

**Reply by stefansundin** [reply]
Tack för ditt stöd. Det är kanske dags att ta en titt på den sidan igen.

**Reply by stefansundin** [reply]
@ingemar1961 v0.3.6 har nu stöd för dr.dk. Det finns nog buggar, så om du hittar någon så rapportera gärna i #29. Tack!
